bookshelf/resources/views/list.php

```diff
@@ -12,48 +13,42 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#}
-
-{% extends "base.html.twig" %}
-
-{% block content %}
+ob_start()
+?>
 
 <h3>Books</h3>
 <a href="/books/add" class="btn btn-success btn-sm">
   <i class="glyphicon glyphicon-plus"></i>
   Add book
 </a>
 
-{# [START book_list] #}
-{% for book in books %}
+<?php // [START book_list] ?>
+<?php foreach ($books as $i => $book): ?>
 <div class="media">
-  <a href="/books/{{book.id}}">
-    {# [START book_image] #}
-    <div class="media-left">
-      {% if book.image_url %}
-        <img src="{{book.image_url}}">
-      {% else %}
-        <img src="http://placekitten.com/g/128/192">
-      {% endif %}
-    </div>
-    {# [END book_image] #}
+  <a href="/books/<?= $book->id() ?>">
+    <?php // [START book_image] ?>
+    <?php if (!empty($book['image_url'])): ?>
```

Comment:
Yes. I did it because when creating a new book, there's no way to do an empty snapshot. I agree it was a bit confusing, so now, everything uses `->get('field')` for custom fields, `->id()` for ID (because that's a special field), and `$book ? $book->get('field') : ''` for the form values.